src/vg/civcraft/mc/citadel/listener/EntityListener.java

```diff
@@ -154,6 +179,271 @@ public void PlayerPromoteEvent(PromotePlayerEvent event){
     			
     		}
     	}
-    }   	
- 
+    }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void entityInteration(EntityDamageEvent event){
+    	
+    }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void hangingPlaceEvent(HangingPlaceEvent event){
+		Player p = event.getPlayer();
+		Block b = event.getBlock().getRelative(event.getBlockFace());
+		Inventory inv = p.getInventory();
+		PlayerState state = PlayerState.get(p);
+		if (ReinforcementMode.REINFORCEMENT_FORTIFICATION != state.getMode()) {
+			return;
+		}
+		if (!canPlace(b, p)){
+			sendAndLog(p, ChatColor.RED, "Cancelled block place, mismatched reinforcement.");
+			event.setCancelled(true);
+			return;
+		}
+		ReinforcementType type = state.getReinforcementType();
+		 // Don't allow double reinforcing reinforceable plants
+        if (wouldPlantDoubleReinforce(b)) {
+        	sendAndLog(p, ChatColor.RED, "Cancelled block place, crop would already be reinforced.");
+            event.setCancelled(true);
+            return;
+        }
+        int required = type.getRequiredAmount();
+        if (type.getItemStack().isSimilar(p.getItemInHand())){
+        	required++;
+        }
+		if (inv.containsAtLeast(type.getItemStack(), required)) {
+			try {
+				if (createPlayerReinforcement(p, state.getGroup(), b, type, p.getItemInHand()) == null) {
+					sendAndLog(p, ChatColor.RED, String.format("%s is not a reinforcible material ", b.getType().name()));
+				} else {
+					state.checkResetMode();
+				}	
+			} catch(ReinforcemnetFortificationCancelException ex){
+				Citadel.getInstance().getLogger().log(Level.WARNING, "ReinforcementFortificationCancelException occured in BlockListener, BlockPlaceEvent ", ex);
+			}
+        } else {
+        	sendAndLog(p, ChatColor.YELLOW, String.format("%s depleted, left fortification mode ",  
+            		state.getReinforcementType().getMaterial().name()));
+            state.reset();
+            event.setCancelled(true);
+        }
+    }
+    
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void hangingEntityBreakEvent(HangingBreakByEntityEvent event){
+    	Reinforcement rein = rm.getReinforcement(event.getEntity().getLocation()); if (rein == null){return;}
```

Comment:
It wasn't deemed necessary. getReinforcement(Block) just reroutes to getReinforcement(Location) anyways.
https://github.com/Civcraft/Citadel/blob/master/src/vg/civcraft/mc/citadel/ReinforcementManager.java#L147
https://github.com/Civcraft/Citadel/blob/master/src/vg/civcraft/mc/citadel/ReinforcementManager.java#L119